sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/grouping.scala

```diff
@@ -32,8 +32,6 @@ trait BaseGroupingSets extends Expression with CodegenFallback {
   def selectedGroupByExprs: Seq[Seq[Expression]]
 
   def groupByExprs: Seq[Expression] = {
-    assert(children.forall(_.resolved),
-      "Cannot call BaseGroupingSets.groupByExprs before the children expressions are all resolved.")
```

Comment:
why do we remove this assert? The following code calls `semanticEquals` and we may get wrong if children are not resolved.

Reply:
> why do we remove this assert? The following code calls `semanticEquals` and we may get wrong if children are not resolved.

With this check it will failed. Hmm I'm still checking the analysis sequence of the analyzer to see why this happens.

Reply:
Have you figured out it?

Reply:
> Have you figured out it?

I got it, there is some unresolved function and alias such as query 
`SELECT a + b, b, SUM(a - b) FROM testData GROUP BY a + b, b WITH CUBE;` , `a+b` is unresolved
`SELECT a + b AS k1, b AS k2, SUM(a - b) FROM testData GROUP BY CUBE(k1, k2);` `k1`, and `k1` is unresolved alias.

And we can see the order `ResolveGroupingAnalytics` is called before `ResolveFunction` and `ResolveAlias`

Reply:
It's fragile to rely on rule order. `ResolveGroupingAnalytics` should only be triggered after expressions are all resolved.

Reply:
Let me make it clear: we must keep the assert here, it's wrong to call `semanticEquals` if the expression is not resolved yet.

Reply:
Fix this